File: src/main/java/com/paranoidkiwi/chocolate/utils/commoncomponents/ui/UIComponent.java
package com.paranoidkiwi.chocolate.utils.commoncomponents.ui;

import com.paranoidkiwi.chocolate.core.appearances.Appearance;
import com.paranoidkiwi.chocolate.core.bounds.NoBoundingBox;
import com.paranoidkiwi.chocolate.core.components.CollisionableComponent;
import com.paranoidkiwi.chocolate.core.reactions.annotations.OnCustomEvent;

public abstract class UIComponent extends CollisionableComponent {

	public static final int UI_COMPONENT_CG = 7772;
	public static final String MAIN_ACTION = "UI Main Action";

	// ****************************************************************
	// ** CONSTRUCTORS
	// ****************************************************************

	public UIComponent(Appearance appearance, double x, double y) {
		super(new NoBoundingBox(), UI_COMPONENT_CG, x, y);

		this.setAppearance(appearance);
	}

	// ****************************************************************
	// ** TRIGGERS
	// ****************************************************************

	@OnCustomEvent(MAIN_ACTION)
	public abstract void performMainAction();
}

File: src/main/java/com/paranoidkiwi/chocolate/utils/commoncomponents/ui/UIComponent.scala
package com.paranoidkiwi.chocolate.utils.commoncomponents.ui

import com.paranoidkiwi.chocolate.core.appearances.Appearance
import com.paranoidkiwi.chocolate.core.collisions.NoBoundingBox
import com.paranoidkiwi.chocolate.core.components.CollisionableComponent
import com.paranoidkiwi.chocolate.core.dimensions.Position
import com.paranoidkiwi.chocolate.core.reactions.annotations.OnCustomEvent

object UIComponent {
	final val UI_COMPONENT_CG = 7772
	final val MAIN_ACTION = "UI Main Action"
}

import UIComponent._

abstract class UIComponent(position: Position, appearance: Appearance) 
	extends CollisionableComponent(position, appearance, NoBoundingBox, UI_COMPONENT_CG) {
  

	// ****************************************************************
	// ** TRIGGERS
	// ****************************************************************

	@OnCustomEvent(MAIN_ACTION)
	def performMainAction
}